libvcx/src/aries/handlers/proof_presentation/verifier/state_machine.rs

```diff
@@ -122,8 +108,7 @@ impl VerifierSM {
                                     ProblemReport::create()
                                         .set_comment(err.to_string())
                                         .set_thread_id(&state.presentation_request.id.0);
-
-                                connection::send_message(connection_handle, problem_report.to_a2a_message())?;
+                                send_message.ok_or(VcxError::from_msg(VcxErrorKind::InvalidState, "Attempted to call undefined send_message callback"))?(&problem_report.to_a2a_message())?;
```

Comment:
Comes down to personal preference but I find this hard to read. I'd prefer
```
let send_message = send_message.ok_or(VcxError::from_msg(VcxErrorKind::InvalidState, "Attempted to call undefined send_message callback"))?;
send_message(&problem_report.to_a2a_message())?;
```
I'll leave it up to you, anyway as I mentioned in private discussion, I think eventually the side effect of sending response should IMO be pulled out from Aries FSMs.

Reply:
You are right in that the line is way too long. Even 
```
let send_message = send_message.ok_or(VcxError::from_msg(VcxErrorKind::InvalidState, "Attempted to call undefined send_message callback"))?;
```
seems too long. How about
```
send_message.ok_or(
    VcxError::from_msg(VcxErrorKind::InvalidState, "Attempted to call undefined send_message callback")
)?(&problem_report.to_a2a_message())?;
```